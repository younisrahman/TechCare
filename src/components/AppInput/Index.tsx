import React from 'react';
import {
  View,
  StyleSheet,
  TextInput,
  StyleProp,
  ViewStyle,
} from 'react-native';
// import { CircleButton } from '@app/components';
// import FontAwsome from 'react-native-vector-icons/FontAwesome';
// import AntDesign from 'react-native-vector-icons/AntDesign';
// import { RFValue } from 'react-native-responsive-fontsize';
import {
  heightPercentageToDP as hp,
  widthPercentageToDP as wp,
} from 'react-native-responsive-screen';
import { Shadow } from '@app/styles/Shadow';
import { H7 } from '@app/styles/typography';
import { Colors, Padding } from '@app/theme';

interface nameStyling {
  titleName?: string;
  style?: StyleProp<ViewStyle>;
  placeHolderText: string;
}

const AppInputDemo: React.FC<nameStyling> = ({
  titleName,
  style,
  placeHolderText,
}) => {
  return (
    <View style={styles.TextInputContainer}>
      <H7 style={styles.textStyling}>{titleName}</H7>
      <TextInput
        style={[styles.firstTextInputStyling, style, Shadow.shadow]}
        placeholder={placeHolderText}
      />
    </View>
  );
};
const styles = StyleSheet.create({
  TextInputContainer: {
    alignSelf: 'center',
    justifyContent: 'center',
    minHeight: hp(7.2),
    width: '100%',
    borderRadius: hp(1.5),
    marginVertical: hp(1),
  },
  textStyling: {
    color: Colors.primaryPurple,
    fontFamily: 'Inter',
  },
  firstTextInputStyling: {
    minHeight: hp(5),
    width: '100%',
    borderRadius: wp(2),
    backgroundColor: Colors.light3,
    paddingHorizontal: Padding.paddingHorizontal,
    marginVertical: hp(1),
  },
});
export default AppInputDemo;
