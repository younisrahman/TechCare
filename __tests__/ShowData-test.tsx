import React from 'react';
import renderer from 'react-test-renderer';
import Index from '../src/components/ShowData/index';

test('renders correctly', () => {
  const tree = renderer
    .create(<Index title="hi" info="ook" margintop={10} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
