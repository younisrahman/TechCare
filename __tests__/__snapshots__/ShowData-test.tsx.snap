// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginTop": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#000",
        "fontFamily": "Rubik-Bold",
      }
    }
  >
    hi
     :
     
  </Text>
  <Text
    style={
      Object {
        "color": "#000",
        "fontFamily": "Rubik-Medium",
      }
    }
  >
    ook
  </Text>
</View>
`;
